registry/registry.go

```diff
@@ -123,6 +125,23 @@ func (r *RouteRegistry) register(uri route.Uri, endpoint *route.Endpoint) route.
 	}
 
 	endpointAdded := pool.Put(endpoint)
+	//Check endpoint for a load balancing algorithm. If it does exist & differs from that of the pool, then take the lb algorithm of the endpoint, otherwise keep the pools lb algorithm.
+	if len(endpoint.LoadBalancingAlgorithm) > 0 && strings.Compare(endpoint.LoadBalancingAlgorithm, pool.LBAlgorithm) != 0 {
```

Comment:
From the [strings.Compare](https://pkg.go.dev/strings#Compare) documentation:

> Compare is included only for symmetry with package bytes. It is usually clearer and **always faster** to use the built-in string comparison operators ==, <, >, and so on.

(emphasis mine)

Reply:
done